### PR TITLE
feat(ci): improve protoc install & add code coverage checks

### PR DESCRIPTION
**Pull request by Arqu**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -261,6 +262,70 @@ jobs:
...
+        echo "$HOME/.local/bin" >> $GITHUB_PATH
+ 
+    - name: coverage
+      timeout-minutes: 30
+      env:
+        # iroh-util/src/lock.rs#test_locks uses a fork() command & fails only in
+        # GHA if this is not present
+        # https://twitter.com/steipete/status/921066430220652544?lang=en
+        # https://blog.phusion.nl/2017/10/13/why-ruby-app-servers-break-on-macos-high-sierra-and-what-can-be-done-about-it/
+        OBJC_DISABLE_INITIALIZE_FORK_SAFETY: "YES"
```
Comment (b5) [inline .github/workflows/ci.yml:320]: Have we run without this on our new runners to see if macOS still chokes?

File: xtask/Cargo.toml [hunk 2]
```diff
@@ -10,4 +10,5 @@ anyhow = "1"
 clap = { version = "4.0.9", features = ["derive"] }
 clap_mangen = "0.2.2"
 dirs-next = "2.0.0"
+xtaskops = "0.2.2"
```
Comment (b5) [inline xtask/Cargo.toml:13]: Neato!

**Review by b5 — APPROVED** [review]
Lgtm

**Comment by b5** [comment]
Can we add the codecov configuration file that turns off the report-comment-on-PR thingie? It's super irritating

**Review by flub — DISMISSED** [review]

**Review by flub — APPROVED** [review]

**Pull request merged**